src/satellite/config.ts

```diff
@@ -34,74 +34,88 @@ export const satelliteDefaults: SatelliteOpts = {
   minSnapshotWindow: 40
 }
 
-export interface SatelliteClientOpts {
-  app: string
-  env: string
-  token: string
-  port: number
-  address: string
-  timeout: number
-  pushPeriod: number
-}
-
 export const satelliteClientDefaults = {
+  env: "default",
   timeout: 3000,
-  pushPeriod: 500
+  pushPeriod: 500,
+  insecure: false
 }
 
-export interface SatelliteClientOverrides {
+const baseDomain = "electric-sql.com"
+
+export interface SatelliteClientOpts {
   app: string
-  env: string
+  env?: string
   token: string
+  host: string
   port: number
-  address: string
   timeout?: number
   pushPeriod?: number
+  insecure?: boolean
 }
 
+
 // Config spec
 export interface ElectricConfig {
   app: string
-  env: string
+  env?: string
   token: string
   migrations?: Migration[],
   replication?: {
-    address: string
+    host: string
     port: number
+    insecure: boolean
   }
   debug?: boolean,
 }
 
+const electricConfigDefaults: Partial<ElectricConfig> = {
+  env: "default"
+}
+
+export const addDefaultsToElectricConfig = (config: ElectricConfig) => {
```

Comment:
This should have an explicit return type, since it's easy to break the contract right now by changing a little bit of code

Reply:
done (tried that at some point, but code was different and not possible then)